### PR TITLE
Attempt to fix #5974 - capsule installation needs RHSCL

### LINKED ISSUE #5974: capsule install requires SCL repos again
SSIA

### PR DESCRIPTION
**Pull request by jhutar**

### CODE REVIEW

**Comment by jhutar** [comment]
At Fry I was able to reproduce the issue:

```
==================================== ERRORS ====================================
 ERROR at setup of CapsuleContentManagementTestCase.test_positive_insights_puppet_package_availability 

cls = <class 'tests.foreman.api.test_contentmanagement.CapsuleContentManagementTestCase'>

    @classmethod
    @skip_if_not_set('capsule', 'clients', 'fake_manifest')
    def setUpClass(cls):
        """Create a separate capsule for tests"""
        super(CapsuleContentManagementTestCase, cls).setUpClass()
        cls.capsule_vm = CapsuleVirtualMachine()
>       cls.capsule_vm.create()

tests/foreman/api/test_contentmanagement.py:138: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
robottelo/vm_capsule.py:251: in create
    self._setup_capsule()
robottelo/vm_capsule.py:240: in _setup_capsule
    location_ids=self._capsule_location_ids
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

capsule_vm = <robottelo.vm_capsule.CapsuleVirtualMachine object at 0x7f27c4b59048>
org_id = '129', lce_id = '175', organization_ids = [], location_ids = []

    def setup_capsule_virtual_machine(capsule_vm, org_id=None, lce_id=None,
                                      organization_ids=None, location_ids=None):
        """Setup a Virtual Machine to host a capsule node
    
        :param capsule_vm: the virtual machine to
         setup as a capsule node
        :param org_id: The organization that setup the capsule content
        :param lce_id: The lifecycle environment  the capsule content was
         promoted.
        :param organization_ids: the organization ids of
         organizations that will use the capsule.
        :param location_ids: the location ids for which the content
         will be synchronized.
        :return tuple: capsule, org, lce  objects
    
        Notes:
    
        1. as this setup need the default manifest to be consumed, please ensure
           that this function is called from one thread.
    
        2. The org will consume the default manifest, it probable that it cannot
           be installed for other organizations, if the tests need a manifest to be
           uploaded, use the same org that setup the capsule by providing the
           org_id and lce_id to be able to create and promote the capsule
           content view and to create the capsule subscription key.
        """
        distro = capsule_vm.capsule_distro
        if distro not in (DISTRO_RHEL7,):
            raise CLIFactoryError(
                u'virtual machine distro "{}" not supported'.format(distro)
            )
    
        # Get the necessary repositories info to setup a capsule host
        capsule_vm_distro_repos = _get_capsule_vm_distro_repos(distro)
        rh_product_arch, rh_product_releasever, rh_repos = capsule_vm_distro_repos
    
        if organization_ids is None:
            organization_ids = []
    
        if location_ids is None:
            location_ids = []
    
        if org_id is None:
            # create a new org
            capsule_org = make_org({
                'name': 'capsule-{0}'.format(gen_string('alpha', 10))})
            org_id = capsule_org['id']
        else:
            capsule_org = Org.info({'id': org_id})
    
        if lce_id is None:
            # Create a new lifecycle environment for capsule only
            capsule_lce = make_lifecycle_environment({
                'name': 'capsule-{0}'.format(gen_string('alpha', 10)),
                'organization-id': org_id
            })
            lce_id = capsule_lce['id']
        else:
            capsule_lce = LifecycleEnvironment.info({
                'id': lce_id, 'organization-id': org_id})
    
        content_setup_data = setup_cdn_and_custom_repos_content(
            org_id=org_id,
            lce_id=lce_id,
            repos=rh_repos,
            upload_manifest=True,
            rh_subscriptions=[
                DEFAULT_SUBSCRIPTION_NAME,
                SATELLITE_SUBSCRIPTION_NAME,
            ]
        )
        setup_virtual_machine(
            capsule_vm,
            capsule_org['label'],
            activation_key=content_setup_data['activation_key']['name'],
            patch_os_release_distro=capsule_vm.capsule_distro,
            rh_repos_id=[
                repo['repository-id']
                for repo in rh_repos if repo['cdn']
            ],
            install_katello_agent=False,
        )
        # Refresh the subscription
        capsule_vm.run('subscription-manager refresh')
        capsule_vm.run('yum clean all && yum repolist')
    
        cert_file_path = '/tmp/{0}-certs.tar'.format(capsule_vm.hostname)
        result = ssh.command(
            'capsule-certs-generate '
            '--foreman-proxy-fqdn {0} '
            '--certs-tar {1}'
            .format(capsule_vm.hostname, cert_file_path)
        )
        if result.return_code != 0:
            raise CLIFactoryError(
                u'was unable to generate certificate\n{}'.format(result.stderr))
    
        # retrieve the installer command from the result output
        satellite_installer_cmd = extract_capsule_satellite_installer_command(
            result.stdout
        )
        # copy the certificate to capsule vm
        _, temporary_local_cert_file_path = mkstemp(suffix='-certs.tar')
        download_file(
            remote_file=cert_file_path,
            local_file=temporary_local_cert_file_path,
            hostname=settings.server.hostname
        )
        upload_file(
            local_file=temporary_local_cert_file_path,
            remote_file=cert_file_path,
            hostname=capsule_vm.hostname
        )
        # delete the temporary file
        os.remove(temporary_local_cert_file_path)
        # Install Satellite Capsule product
        capsule_vm.run('yum install -y satellite-capsule')
        result = capsule_vm.run('rpm -q satellite-capsule')
        if result.return_code != 0:
            raise CLIFactoryError(
                u'Failed to install satellite-capsule package\n{}'.format(
>                   result.stderr)
            )
E           robottelo.cli.factory.CLIFactoryError: Failed to install satellite-capsule package
E           b''

robottelo/cli/factory.py:3775: CLIFactoryError
============================== 0 tests deselected ==============================
========================== 1 error in 2769.91 seconds ==========================
```

but now I'm unable to test the fix, as I'm getting:

```
==================================== ERRORS ====================================
 ERROR at setup of CapsuleContentManagementTestCase.test_positive_insights_puppet_package_availability 

cls = <class 'tests.foreman.api.test_contentmanagement.CapsuleContentManagementTestCase'>

    @classmethod
    @skip_if_not_set('capsule', 'clients', 'fake_manifest')
    def setUpClass(cls):
        """Create a separate capsule for tests"""
        super(CapsuleContentManagementTestCase, cls).setUpClass()
>       cls.capsule_vm = CapsuleVirtualMachine()

tests/foreman/api/test_contentmanagement.py:137: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
robottelo/vm_capsule.py:64: in __init__
    server_host_os_version = get_host_os_version()
robottelo/host_info.py:17: in get_host_os_version
    cmd = ssh.command('cat /etc/redhat-release')
robottelo/ssh.py:285: in command
    timeout=connection_timeout) as connection:
/usr/lib64/python3.6/contextlib.py:81: in __enter__
    return next(self.gen)
robottelo/ssh.py:148: in get_connection
    hostname, username, password, key_filename, timeout
robottelo/ssh.py:107: in get_client
    timeout=timeout
venv/lib/python3.6/site-packages/paramiko/client.py:338: in connect
    retry_on_signal(lambda: sock.connect(addr))
venv/lib/python3.6/site-packages/paramiko/util.py:279: in retry_on_signal
    return function()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   retry_on_signal(lambda: sock.connect(addr))
E   socket.timeout: timed out

venv/lib/python3.6/site-packages/paramiko/client.py:338: timeout
============================== 0 tests deselected ==============================
=========================== 1 error in 10.30 seconds ===========================
```

File: robottelo/constants.py [hunk 1]
```diff
@@ -245,6 +246,8 @@
     'rhaht': 'Red Hat Enterprise Linux Atomic Host (Trees)',
     'rhdt7': ('Red Hat Developer Tools RPMs for Red Hat Enterprise Linux 7'
               ' Server'),
+    'rhscl7': ('Red Hat Software Collections RPMs for Red Hat Enterprise'
+              ' Linux 7 Server'),
```
Comment (JacobCallahan) [inline robottelo/constants.py]: add a single space in front of the first ' to align these for flake.

Comment (jhutar) [reply robottelo/constants.py]: Added some spaces here and there ;-) Should be better now. I have not tested it successfully yet still.

**Comment by ldjebran** [comment]
Please provide test results one of the tests using capsule content https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/cli/test_contentview.py#L4213
or here capsule is installed at setup
https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/api/test_contentmanagement.py#L127

**Comment by jhutar** [comment]
Roman did it in https://github.com/SatelliteQE/robottelo/pull/6130/

**Pull request closed without merging**